lighthouse-core/gather/gatherers/seo/font-size.js

```diff
@@ -109,142 +184,175 @@ function getNodeTextLength(node) {
 /**
  * @param {Driver} driver
  * @param {LH.Crdp.DOM.Node} node text node
- * @returns {WebInspector.CSSStyleDeclaration}
+ * @returns {Promise<FailingNodeData['cssRule']|undefined>}
  */
-function getFontSizeSourceRule(driver, node) {
-  return driver.sendCommand('CSS.getMatchedStylesForNode', {nodeId: node.nodeId})
-    .then(matchedRules => getEffectiveRule(FONT_SIZE_PROPERTY_NAME, node, matchedRules));
+async function fetchSourceRule(driver, node) {
+  const matchedRules = await driver.sendCommand('CSS.getMatchedStylesForNode', {
+    nodeId: node.nodeId,
+  });
+  const sourceRule = getEffectiveFontRule(matchedRules);
+  if (!sourceRule) return undefined;
+
+  return {
+    type: sourceRule.type,
+    range: sourceRule.range,
+    styleSheetId: sourceRule.styleSheetId,
+    parentRule: sourceRule.parentRule && {
+      origin: sourceRule.parentRule.origin,
+      selectors: sourceRule.parentRule.selectors,
+    },
+  };
 }
 
 /**
  * @param {Driver} driver
  * @param {LH.Artifacts.FontSize.DomNodeWithParent} node text node
- * @returns {Promise<?{fontSize: number, textLength: number, node: LH.Artifacts.FontSize.DomNodeWithParent}>}
+ * @returns {Promise<?FailingNodeData>}
  */
-function getFontSizeInformation(driver, node) {
-  return driver.sendCommand('CSS.getComputedStyleForNode', {nodeId: node.parentId})
-    .then(result => {
-      const {computedStyle} = result;
-      const fontSizeProperty = computedStyle.find(({name}) => name === FONT_SIZE_PROPERTY_NAME);
-
-      return {
-        fontSize: parseInt(fontSizeProperty.value, 10),
-        textLength: getNodeTextLength(node),
-        node: node.parentNode,
-      };
-    })
-    .catch(err => {
-      require('../../../lib/sentry.js').captureException(err);
-      return null;
+async function fetchComputedFontSize(driver, node) {
+  try {
+    const {computedStyle} = await driver.sendCommand('CSS.getComputedStyleForNode', {
+      nodeId: node.parentId,
     });
+
+    const fontSizeProperty = computedStyle.find(({name}) => name === FONT_SIZE_PROPERTY_NAME);
+
+    return {
+      // @ts-ignore - font size property guaranteed to be returned in getComputedStyle
+      fontSize: parseInt(fontSizeProperty.value, 10),
+      textLength: getNodeTextLength(node),
+      node: /** @type {LH.Artifacts.FontSize.DomNodeWithParent} */ (node.parentNode),
+    };
+  } catch (err) {
+    Sentry.captureException(err);
```

Comment:
should we debounce this in keeping with #5994?

Reply:
thoughts on #6215 first? :)